### PR TITLE
[SPARK-23918][SQL] Add array_min function

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

The PR adds the SQL function `array_min`. It takes an array as argument and returns the minimum value in it.

## How was this patch tested?

added UTs

### CODE REVIEW

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -2080,6 +2080,21 @@ def size(col):
...
+def array_min(col):
+    """
+    Collection function: returns the minimum value of the array.
+
+    :param col: name of column or expression
+
+    >>> df = spark.createDataFrame([([2, 1, 3],), ([None, 10, -1],)], ['data'])
+    >>> df.select(array_min(df.data).alias('min')).collect()
+    [Row(min=1), Row(min=-1)]
+     """
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py]: quick nit

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala [hunk 2]
```diff
@@ -287,3 +287,70 @@ case class ArrayContains(left: Expression, right: Expression)
...
+/**
+ * Returns the minimum value in the array.
+ */
+@ExpressionDescription(
+usage = "_FUNC_(array) - Returns the minimum value in the array.",
+examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 20, null, 3));
+       1
+  """, since = "2.4.0")
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala]: indentation ..

File: python/pyspark/sql/functions.py (hunk 1 above)
Comment (mgaido91) [reply python/pyspark/sql/functions.py]: sorry, I can't see what is the problem here. May you please clarify? Thanks.

Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py]: """ seems having one more leading space ..

Comment (mgaido91) [reply python/pyspark/sql/functions.py]: you are right, good catch! I was looking for reference at the `sort_array` function below which has the same issue. I will fix it there too, thanks.

**Comment by gatorsmile** [comment]
cc @ueshin

**Comment by kiszk** [comment]
retest this please

**Comment by mgaido91** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala [hunk 3]
```diff
@@ -287,3 +287,67 @@ case class ArrayContains(left: Expression, right: Expression)
 
   override def prettyName: String = "array_contains"
 }
+
+
+/**
+ * Returns the minimum value in the array.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(array) - Returns the minimum value in the array.",
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala]: We will ignore NULL, right? Please document it.

**Comment by gatorsmile** [comment]
@mgaido91 Could you address the conflicts? Thanks!

**Comment by ueshin** [comment]
Thanks! merging to master.

**Pull request closed without merging**